### PR TITLE
arch: arm: arch32: initial proposition for Zero Latency IRQ feature for ARMv6_M

### PR DESCRIPTION
**Pull request by f-egg**
Also, have a look at the following RFC:
Zero-latency interrupt extension for ARMv6-M https://github.com/zephyrproject-rtos/zephyr/issues/32124

Signed-off-by: egg fabio <fegg@baumer.com>

### CODE REVIEW

**Comment by f-egg** [comment]
Hello all, has anyone been able to deal with this topic?

**Comment by carlescufi** [comment]
@ioannisg could you please review this?

File: arch/arm/core/aarch32/irq_manage.c [hunk 1]
```diff
@@ -36,6 +36,90 @@ extern void z_arm_reserved(void);
...
+{
+	zli_shadow_reg = new_value;
+}
+
+uint32_t zli_get_mask(void)
+{
+	return zli_mask;
+}
+
+bool zli_is_zli(unsigned int irq)
```
Comment (bbolen) [inline arch/arm/core/aarch32/irq_manage.c:60]: static

File: arch/arm/core/aarch32/irq_manage.c [hunk 2]
```diff
@@ -36,6 +36,90 @@ extern void z_arm_reserved(void);
 #define REG_FROM_IRQ(irq) (irq / NUM_IRQS_PER_REG)
 #define BIT_FROM_IRQ(irq) (irq % NUM_IRQS_PER_REG)
 
+#ifdef CONFIG_ZERO_LATENCY_IRQS_ARMV6_M
+/* initialise zero latency irq variables*/
+uint32_t zli_shadow_reg;
```
Comment (bbolen) [inline arch/arm/core/aarch32/irq_manage.c:41]: these 3 variables can be marked static

Comment (f-egg) [reply arch/arm/core/aarch32/irq_manage.c:41]: Thank you for the review. I will apply the suggested changes.

**Comment by f-egg** [comment]
@ioannisg and @carlescufi 
I noticed that there are some conflicts that have to be resolved and I am glad to do that. But I cannot invest more time if there is no interest in this topic. Except for the recommended changes by @bbolen, there was no review. I cannot invest more time to resolve the conflicts and then wait again without any movement on this topic. If there still is interest in the proposed solution, I will resolve the conflicts and hope for a more lively discussion. 
Regards Fabio

**Comment by ioannisg** [comment]
> @ioannisg and @carlescufi
> I noticed that there are some conflicts that have to be resolved and I am glad to do that. But I cannot invest more time if there is no interest in this topic. Except for the recommended changes by @bbolen, there was no review. I cannot invest more time to resolve the conflicts and then wait again without any movement on this topic. If there still is interest in the proposed solution, I will resolve the conflicts and hope for a more lively discussion.
> Regards Fabio

Apologies for down-prioritizing this review, @f-egg . I've been busy with 2.6 release related tasks. I am going to take a look as soon as possible.

**Comment by carlescufi** [comment]
@f-egg would you like to resuscitate this PR? I can probably help you drive it forward.

**Comment by marnold-b** [comment]
@carlescufi I took the branch from @f-egg over, he is not anymore at the company. The feature has at the moment a lower priority for us, but we are still interested in this feature. Any feedback if the implementation is the right approach, is very welcome.
The branch needs to be rebased, I can have a look at that probably next week.

So yes we could resuscitate this PR, if there is interest in this.

**Comment by carlescufi** [comment]
> @carlescufi I took the branch from @f-egg over, he is not anymore at the company. The feature has at the moment a lower priority for us, but we are still interested in this feature. Any feedback if the implementation is the right approach, is very welcome. The branch needs to be rebased, I can have a look at that probably next week.
> 
> So yes we could resuscitate this PR, if there is interest in this.

Thank you @marnold-b. We'd certainly appreciate it, yes, I think there's enough interest in this to make it worth reviving. I cannot actually reopen this PR, but you could create a new one with the commit in this one as a starting point.

**Pull request closed without merging**